src/ophys_etl/transforms/classification.py

```diff
@@ -0,0 +1,304 @@
+    )
+    traces_path = H5InputFile(
+        required=True,
+        description=(
+            "Path to traces extracted from an experiment (h5 format). "
+            "The order of the traces in the dataset should correspond to "
+            "the order of masks in `roi_masks_path`.")
+    )
+    traces_data_key = fields.Str(
+        required=False,
+        missing="data",
+        default="data",
+        description=("Key in `traces_path` h5 file where data array is "
+                     "stored.")
+    )
+    roi_masks_path = fields.InputFile(
+        required=True,
+        description=("Path to json file of segmented ROI masks. The file "
+                     "records must conform to the schema "
+                     "`RoiJsonRecordSchema`")
+    )
+    rig = fields.Str(
+        required=True,
+        description=("Name of the ophys rig used for the experiment.")
+    )
+    depth = fields.Int(
+        required=True,
+        description=("Imaging depth for the experiment.")
+    )
+    full_genotype = fields.Str(
+        required=True,
+        description=("Genotype of the experimental subject.")
+    )
+    targeted_structure = fields.Str(
+        required=True,
+        description=("Name of the brain structure targeted by imaging.")
+    )
+    classifier_model_path = fields.Str(
+        required=True,
+        description=("Path to model. Can either be an s3 location or a "
+                     "path on the local file system. The output of the model "
+                     "should be 0 if the ROI is classified as not a cell, "
+                     "and 1 if the ROI is classified as a cell.")
+    )
+    trace_sampling_fps = fields.Int(
+        required=True,
+        description="Sampling rate of trace (frames per second)."
+    )
+    downsample_to = fields.Int(
+        required=False,
+        default=4,
+        validate=lambda x: x > 0,
+        description=("Target rate to downsample trace data (frames per "
+                     "second). Will use average bin values for downsampling.")
+    )
+    output_json = fields.OutputFile(
+        required=True,
+        description="Filepath to dump json output."
+    )
+
+    @validates("classifier_model_path")
+    def validate_classifier_model_path(self, uri: str, **kwargs):
+        """ Check to see if file exists (either s3 or local file) """
+        if uri.startswith("s3://"):
+            s3 = boto3.client("s3")
+            parsed = urlparse(uri, allow_fragments=False)
+            try:
+                s3.head_object(
+                    Bucket=parsed.netloc, Key=parsed.path.lstrip("/"))
+            except ClientError as e:
+                if e.response["ResponseMetadata"]["HTTPStatusCode"] == 404:
+                    raise ValidationError(
+                        f"Object at URI {uri} does not exist.")
+                else:
+                    raise e from None
+        else:
+            if not os.path.exists(uri):
```

Comment:
you are only checking if it exists, but the user could not have read access. You could just invoke [this call](https://github.com/AllenInstitute/argschema/blob/afc2b9241176a587e543f71830f2968d7a552cb7/argschema/fields/files.py#L150-L163) to offload to argschema.

Reply:
I don't think that argschema is very pythonic here. I only included them due to convention. If I were not trying to follow argschema convention, I would not make any checks on the input files for a few reasons:

1. You would get an easy-to-read fatal error if the file didn't exist or couldn't be accessed, so there's not a lot of benefit to doing it separately
2. Checking the file when you validate the schema does not guarantee that the file hasn't changed (deleted/updated permissions) by the time it has been accessed by the main program, so ultimately you still are relying on (1)
3. You'll likely load the files early on, and therefore error early on, so there's not a lot of benefit to checking whether you'll get an error first. Contrast this with output dir, which wouldn't throw an error until the very end. The dir could still be deleted/updated in the course of the program which would cause it to fail at the end, but you could potentially save a lot of time by checking first.
4. Argschema uses different methods for checking Windows vs. unix; it reads the whole thing into memory on Windows machines. For files that were not intended to be read in memory at once, this could crash windows programs and be hard to debug since we don't have CI testing on windows.

Reply:
ok